File: frontend/src/api/avatar.ts
// ============================================================
// 第二分身 API（二期 P1）
// 长期记忆 + 灵感工作流 + 五档配置
// ============================================================
import { http } from './http'
import type { PaginatedData } from '@/types'

export interface AvatarMemory {
  id: string
  user_id: string
  memory_type: 'profile' | 'knowledge' | 'event' | 'relation'
  category: string | null
  title: string
  content: string | null
  tags: string[]
  source: string | null
  confidence: number
  is_verified: boolean
  importance: number
  created_at: string
  updated_at: string
}

export interface AvatarInspiration {
  id: string
  user_id: string
  title: string
  description: string | null
  domain: string | null
  estimated_value: string | null
  prompt: string | null
  result: string | null
  status: 'generated' | 'selected' | 'executing' | 'completed' | 'discarded'
  feedback: string | null
  batch_id: string | null
  source_type: string
  created_at: string
  updated_at: string
}

export interface AvatarConfig {
  id: string
  user_id: string
  automation_level: string
  local_model_enabled: boolean
  local_model_provider: string | null
  local_model_name: string | null
  local_model_url: string | null
  cloud_model_enabled: boolean
  cloud_model_provider: string | null
  cloud_model_name: string | null
  persona_name: string
  persona_setting: string | null
  inspiration_enabled: boolean
  inspiration_frequency: string
  inspiration_domains: string[]
  reply_length: string
  language_style: string
  creativity: number
  operation_overrides: Record<string, unknown>
  updated_at: string
}

export const avatarApi = {
  // 记忆
  listMemories(params?: { memory_type?: string; page?: number; page_size?: number }) {
    return http.get<PaginatedData<AvatarMemory>>('/avatar/memories', params as Record<string, unknown>)
  },
  memoryStats() {
    return http.get<{ profile: number; knowledge: number; event: number; relation: number; total: number; verified: number }>('/avatar/memories/stats')
  },
  createMemory(data: Partial<AvatarMemory>) {
    return http.post<AvatarMemory>('/avatar/memories', data)
  },
  updateMemory(id: string, data: Partial<AvatarMemory>) {
    return http.patch<AvatarMemory>(`/avatar/memories/${id}`, data)
  },
  deleteMemory(id: string) {
    return http.delete<void>(`/avatar/memories/${id}`)
  },
  verifyMemory(id: string, verified = true) {
    return http.post<AvatarMemory>(`/avatar/memories/${id}/verify?verified=${verified}`)
  },

  // 灵感
  generateInspirations(data: { domain?: string; count?: number; source_type?: string }) {
    return http.post<{ batch_id: string; inspirations: AvatarInspiration[]; count: number }>('/avatar/inspirations/generate', data)
  },
  listInspirations(params?: { status?: string; page?: number; page_size?: number }) {
    return http.get<PaginatedData<AvatarInspiration>>('/avatar/inspirations', params as Record<string, unknown>)
  },
  selectInspiration(id: string) {
    return http.post<AvatarInspiration>(`/avatar/inspirations/${id}/select`)
  },
  executeInspiration(id: string, result?: string) {
    const qs = result ? `?result=${encodeURIComponent(result)}` : ''
    return http.post<AvatarInspiration>(`/avatar/inspirations/${id}/execute${qs}`)
  },
  feedbackInspiration(id: string, feedback: string) {
    return http.post<AvatarInspiration>(`/avatar/inspirations/${id}/feedback?feedback=${feedback}`)
  },
  discardInspiration(id: string) {
    return http.post<AvatarInspiration>(`/avatar/inspirations/${id}/discard`)
  },

  // 配置
  getConfig() {
    return http.get<AvatarConfig>('/avatar/config')
  },
  updateConfig(data: Partial<AvatarConfig>) {
    return http.patch<AvatarConfig>('/avatar/config', data)
  },
}

File: backend/app/schemas/avatar.py
# ============================================================
# 第二分身 schema（二期 P1）
# ============================================================
from __future__ import annotations

from datetime import datetime
from pydantic import BaseModel, Field


class AvatarMemoryOut(BaseModel):
    id: str
    user_id: str
    memory_type: str  # profile/knowledge/event/relation
    category: str | None = None
    title: str
    content: str | None = None
    tags: list = []
    source: str | None = None
    source_id: str | None = None
    confidence: float = 1.0
    is_verified: bool = False
    importance: int = 3
    # 身份轴（三期 B）
    identity_id: str | None = None
    created_at: datetime
    updated_at: datetime


class CreateMemoryRequest(BaseModel):
    memory_type: str = Field(pattern="^(profile|knowledge|event|relation)$")
    category: str | None = None
    title: str = Field(min_length=1, max_length=200)
    content: str | None = None
    tags: list = []
    source: str | None = None
    importance: int = Field(ge=1, le=5, default=3)
    # 身份轴（三期 B）：可空 = 未归类
    identity_id: str | None = None


class AvatarInspirationOut(BaseModel):
    id: str
    user_id: str
    title: str
    description: str | None = None
    domain: str | None = None
    estimated_value: str | None = None
    prompt: str | None = None
    result: str | None = None
    status: str  # generated/selected/executing/completed/discarded
    feedback: str | None = None  # useful/useless/normal
    batch_id: str | None = None
    source_type: str = "active"
    created_at: datetime
    updated_at: datetime


class GenerateInspirationRequest(BaseModel):
    domain: str | None = None
    count: int = Field(ge=1, le=10, default=3)
    source_type: str = "active"


class AvatarConfigOut(BaseModel):
    id: str
    user_id: str
    automation_level: str = "L2"
    local_model_enabled: bool = False
    local_model_provider: str | None = None
    local_model_name: str | None = None
    local_model_url: str | None = None
    cloud_model_enabled: bool = True
    cloud_model_provider: str | None = None
    cloud_model_name: str | None = None
    persona_name: str = "启明星"
    persona_setting: str | None = None
    inspiration_enabled: bool = True
    inspiration_frequency: str = "manual"
    inspiration_domains: list = []
    reply_length: str = "medium"
    language_style: str = "professional"
    creativity: float = 0.7
    operation_overrides: dict = {}
    updated_at: datetime


class UpdateConfigRequest(BaseModel):
    automation_level: str | None = None
    local_model_enabled: bool | None = None
    local_model_provider: str | None = None
    local_model_name: str | None = None
    local_model_url: str | None = None
    cloud_model_enabled: bool | None = None
    cloud_model_provider: str | None = None
    cloud_model_name: str | None = None
    cloud_api_key: str | None = None
    persona_name: str | None = None
    persona_setting: str | None = None
    inspiration_enabled: bool | None = None
    inspiration_frequency: str | None = None
    inspiration_domains: list | None = None
    reply_length: str | None = None
    language_style: str | None = None
    creativity: float | None = None
    operation_overrides: dict | None = None
